lib/mongoose-field-encryption.js

```diff
@@ -124,29 +124,37 @@ const fieldEncryption = function(schema, options) {
     }
   });
 
-  schema.pre("update", function(_next) {
+  function updateHook(_next) {
     const next = getCompatitibleNextFunc(_next);
     for (let field of fieldsToEncrypt) {
       let encryptedFieldName = encryptedFieldNamePrefix + field;
-      let encryptedFieldValue = this._update.$set[encryptedFieldName];
-      let plainTextValue = this._update.$set[field];
+      this._update.$set = this._update.$set || {};
+      let plainTextValue = this._update.$set[field] || this._update[field];
 
-      if (encryptedFieldValue === false && plainTextValue) {
+      if (plainTextValue) {
```

Comment:
It looks like the check for `encryptedFieldValue === false` is removed now. This is very necessary as the value might be double encrypted if it is passed to the update statement by mistake.

Reply:
We should probably also add a test for this case :)

Reply:
You are right :) I'll try to cover it with a test somehow